core/invehicle-digital-twin/src/invehicle_digital_twin_config.rs

```diff
@@ -12,7 +12,7 @@ const CONFIG_FILENAME: &str = "invehicle_digital_twin_settings";
 #[derive(Debug, Deserialize)]
 pub struct Settings {
```

Comment:
For mutually exclusive config settings, consider using an enum. The serde library has some nice support for serializing these too: https://serde.rs/enum-representations.html . This avoids the need to remove one of the entries from the config template as mentioned in the readme, guarantees mutual exclusivity, avoids potential confusion for users, and might make the logic for retrieving the uri simpler.

For example, this struct might look like this:

```rust
#[derive(Debug, Deserialize)]
pub struct Settings {
    pub invehicle_digital_twin_target: Target,
}

#[derive(Debug, Deserialize)]
pub enum Target {
    Authority(String),
    ChariottUri(String),
}
```

while config would look like one of the following:

```json
{
  "invehicle_digital_twin_target": {
    "Authority": "<auth>"
  }
}
```

```json
{
  "invehicle_digital_twin_target": {
    "ChariottUri": "<chariott>"
  }
}
```

Reply:
It looks like @jorchiu already left a similar comment elsewhere, but I will leave mine since the examples might be useful and there's some additional justification for why you might want to do this

Reply:
This comes down to personal preferences.  Leaving it as is.